File: backend/src/Organi.Server/Organi.Server.Domain/Entities/Category.cs
using Organi.Server.Domain.Common;

namespace Organi.Server.Domain.Entities;

public sealed class Category : AuditableEntity
{
    public string Name { get; set; } = default!;
    public string Slug { get; set; } = default!;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public int DisplayOrder { get; set; }
    public bool IsActive { get; set; } = true;
    public Guid? ParentCategoryId { get; set; }

    public Category? ParentCategory { get; set; }
    public ICollection<Category> ChildCategories { get; set; } = [];
    public ICollection<Product> Products { get; set; } = [];
}

File: frontend/web/src/types/api/category.ts
// Mirrors Organi.Server.Application.Features.Categories.DTOs.CategoryResponse
export interface CategoryResponse {
  id: string
  name: string
  slug: string
  description: string | null
  imageUrl: string | null
  displayOrder: number
  isActive: boolean
  parentCategoryId: string | null
  children: CategoryResponse[]
}

// Mirrors CreateCategoryCommand — isActive is NOT settable on create (server forces true)
export interface CreateCategoryRequest {
  name: string
  description: string | null
  imageUrl: string | null
  displayOrder: number
  parentCategoryId: string | null
}

// Mirrors UpdateCategoryCommand (Id travels in the URL, not the body)
export interface UpdateCategoryRequest extends CreateCategoryRequest {
  isActive: boolean
}
